WordPress/Classes/ViewRelated/Blog/Site Picker/BlogList/BlogListViewModel.swift

```diff
@@ -23,6 +23,17 @@ final class BlogListViewModel: NSObject, ObservableObject {
 
     var onAddSiteTapped: (AddSiteMenuViewModel.Selection) -> Void = { _ in }
 
+    weak var sidebarViewModel: SidebarViewModel?
+
+    var currentSite: BlogListSiteViewModel? {
+        guard let sidebarViewModel,
+              case let .blog(selectedBlogID) = sidebarViewModel.selection else {
+            return nil
+        }
+
+        return allSites.first { $0.id == selectedBlogID }
```

Comment:
(nit) I'd probably suggest using `try mainContext.existingObject(with: objectID)` instead of iterating over `allSites`.

Reply:
👍 implemented here 5bcfca312002e6c794963900ce22e260631f4225